pages/reference/base-images/base-images.md

```diff
@@ -10,91 +10,90 @@ excerpt: Docker images maintained by {{ $names.company.lower }}
 
 ### How the Image Naming Scheme Works
 
-With over 26000 `balenalib` base images to choose from it can be overwhelming to decide which image and tag is correct for your project. To pick the correct image, it helps to understand how the images are named as that indicates what is installed in the image. In general the naming scheme for the `balenalib` image set follows the pattern below:
+With over 26000 `balenalib` base images to choose from, it can be overwhelming to decide which image and tag are correct for your project. To pick the correct image, it helps to understand how the images are named as that indicates what is installed in the image. In general, the naming scheme for the `balenalib` image set follows the pattern below:
+
 ```
 balenalib/<hw>-<distro>-<lang_stack>:<lang_ver>-<distro_ver>-(build|run)-<yyyymmdd>
 ```
 
-#### Image Names:
+#### Image Names
 
-- `<hw>` is either architecture or device type and is **mandatory**. If using Dockerfile.templates you can replace this with `%%BALENA_MACHINE_NAME%%` or `%%BALENA_ARCH%%`. For a list of available device names and architectures see the [Device types](/reference/base-images/devicetypes/).
-- `<distro>`  is the linux distribution, currently there are 4 distributions, namely Debian, Alpine, Ubuntu and Fedora. This field is optional, and will default to Debian if left out.
-- `<lang_stack>` is the programming language pack, currently we support Node.js, Python, OpenJDK and Go. This field is optional and if left out, no language pack will be installed, so you will just have the distribution.
+- `<hw>` is either architecture or device type and is **mandatory**. If using Dockerfile.templates, you can replace this with `%%BALENA_MACHINE_NAME%%` or `%%BALENA_ARCH%%`. For a list of available device names and architectures, see the [Device types](/reference/base-images/devicetypes/).
+- `<distro>` is the Linux distribution. Currently there are 4 distributions, namely Debian, Alpine, Ubuntu and Fedora. This field is optional and will default to Debian if left out.
+- `<lang_stack>` is the programming language pack, currently we support Node.js, Python, OpenJDK and Go. This field is optional, and if left out, no language pack will be installed, so you will just have the distribution.
```

Comment:
So I guess if you specify a `<lang_ver>` but not a `<lang_stack>`, the lang_ver will just be ignored?

Reply:
It won't be ignored and it'd probably just mean the image doesn't exist (unless it happened to coincide with an available tag) - but either way, it's unlikely to be what you want. For example, you can see the `raspberrypi3-alpine` tags here so if you specified a <lang_ver> tag of 10.8 (e.g. node) then there simply isn't a corresponding image.

Reply:
Cool, I was just checking what the edge-case behaviour was :slightly_smiling_face:

Reply:
Hmmm, that's curious - https://hub.docker.com/layers/balenalib/raspberrypi3-alpine/latest/images/sha256-054582a87ef96ba2243334500bb580246dba066967e9f6ad0c3bee655fde281d says "LABEL io.balena.device-type=raspberry-pi2" ??